Add APP2 launch file

MathieuMRodrigue opened the pull request:
As Alexandre requested, simple launch file removing lidar and camera features for app2 (following rqt_graph requested)

<img width="979" height="333" alt="image" src="https://github.com/user-attachments/assets/58485da2-9073-4cb9-b387-2e067a1baabe" />

MathieuMRodrigue committed:
Add APP2 launch file

As Alexandre requested, simple launch file removing lidar and camera features for app2 (following rqt_graph requested)

chameau5050 commented on racecar_bringup/launch/gro501.launch.py:
There are too many duplicate codes; this launch file mostly does the same thing: the command teleop.launch.py start_lidar:=false start_camera:=false. If we need a launch file specific for this APP, I would just include the teleop.launch.py with the described argument

MathieuMRodrigue replied:
I agree with the duplicating code, but the only reason that i did it that way, is because Alexandre Girard asked us that when he run rqt_graph, it would generate the exact graph that i put on the PR. If i use teleop like you mentionned, the rqt_graph have extra nodes that he doesn't want!

<img width="979" height="333" alt="image" src="https://github.com/user-attachments/assets/5dd247c3-a164-449b-bf3e-be6dcf6d4838" />

MathieuMRodrigue replied:
I will work on another option with less duplicate code !

MathieuMRodrigue replied:
I pushed my modifications (see last commit)

MathieuMRodrigue committed:
Remove arduinoSensor node

MathieuMRodrigue committed:
Launch modifications

This commit include more than the APP2 launch file modification:

-Overall renaming of the launch files (changed with APPx.launch name)

-Modifications for the bringup launch : Add conditions/arugments for every nodes to be able to populate desired nodes.

- Removed duplicating code for the APPx launch file, now it's just launching teleop with desired nodes as argument to acheive Alexandre request.

- Fix wall_estimator /scan remapping to /racecar/scan for APP4 launch.

chameau5050 commented on racecar_bringup/launch/teleop.launch.py:
```diff
@@ -12,14 +12,8 @@ def generate_launch_description():
-            launch_arguments={'start_lidar': LaunchConfiguration('start_lidar'),
-                              'start_camera': LaunchConfiguration('start_camera')}.items()
-        ),
+            PythonLaunchDescriptionSource([os.path.join(racecar_bringup, 'launch', 'bringup.launch.py')])),
```
I would prefer if we keep scoped argument pass for the two following reasons: 

1. It is a lot clearer when you read the file
2. It looks like it's might come by default in next ros release breaking the current code https://github.com/ros2/launch/issues/801

I suggest just reverting this file to fix the issue for this file

chameau5050 commented on racecar_bringup/launch/APP5.launch.py:
I suggest reverting the file name to gro830 since the APP order could change overtime

chameau5050 commented on racecar_bringup/launch/APP4.launch.py:
I suggest reverting the file name to GRO501_P2 or something like that since the APP order could change overtime

chameau5050 commented on racecar_bringup/launch/APP3.launch.py:
I suggest reverting the file name to GRO525 the APP order could change overtime

chameau5050 commented on racecar_bringup/launch/APP2.launch.py:
```diff
@@ -0,0 +1,20 @@
+                              'start_magwick': "False",
+                              'start_kalma': "False"}.items()
+                                )
+                                ])
```
I would prefer if we keep scoped argument pass for the two following reasons: 

1. It is a lot clearer when you read the file
2. It looks like it's might come by default in next ros release breaking the current code https://github.com/ros2/launch/issues/801

chameau5050 commented on racecar_bringup/launch/APP2.launch.py:
I suggest reverting the file name to GRO501_P1 or something like that since the APP order could change overtime

chameau5050 commented on racecar_bringup/launch/APP1.launch.py:
I suggest reverting the file name to GRO520 or something like that since the APP order could change overtime

MathieuMRodrigue committed:
Launches files update

- Rename to suggested names
- Added/Declared launch arguments for every launch files

MathieuMRodrigue reviewed (commented):
- Rename to suggested names
- Added/Declared launch arguments for every launch files

chameau5050 commented on racecar_bringup/launch/GRO525.launch.py:
```diff
@@ -0,0 +1,59 @@
+                              'start_camera': LaunchConfiguration('start_camera'),
+                              'start_magwick': LaunchConfiguration('start_magwick'),
+                              'start_kalma': LaunchConfiguration('start_kalma')}.items()
+        ),
```
In GRO525 we launch bringup seperately to be able to also use it in simulation

chameau5050 commented on racecar_bringup/launch/GRO525.launch.py:
```diff
@@ -0,0 +1,59 @@
+        DeclareLaunchArgument('start_lidar', default_value='False'),
+        DeclareLaunchArgument('start_camera', default_value='True'),
+        DeclareLaunchArgument('start_magwick', default_value='False'),
+        DeclareLaunchArgument('start_kalma', default_value='False'),
```
see comment bellow

chameau5050 commented on racecar_bringup/launch/GRO830.launch.py:
```diff
@@ -1,8 +1,9 @@
 from launch_ros.actions import Node
 from ament_index_python.packages import get_package_share_directory
-from launch.actions import IncludeLaunchDescription
+from launch.actions import IncludeLaunchDescription,DeclareLaunchArgument
```
a bit picky but
```suggestion
from launch.actions import IncludeLaunchDescription, DeclareLaunchArgument
```

chameau5050 reviewed (changes requested):
almost there

MathieuMRodrigue committed:
Remove GRO525 arguments

MathieuMRodrigue reviewed (commented):
Removed arguments in GRO525

chameau5050 reviewed (approved):
Nice job !

MathieuMRodrigue committed:
Small space fix

- Add space on the import ( suggested fix )
